src/components/divider-dropdown-autocomplete/DividerDropdownAutocomplete.tsx

```diff
@@ -0,0 +1,65 @@
+import { SyntheticEvent } from 'react'
+import { useTranslation } from 'react-i18next'
+import { AutocompleteRenderGroupParams } from '@mui/material'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import Divider from '@mui/material/Divider'
+
+import AsyncAutocomplete, {
+  AsyncAutocompleteProps
+} from '~/components/async-autocomlete/AsyncAutocomplete'
+
+import { styles } from '~/containers/my-courses/course-toolbar/CourseToolbar.style'
+
+interface DividedDropdownAutocompleteProps<
+  Response,
+  TransformedResponse,
+  Params
+> extends Omit<
+    AsyncAutocompleteProps<
+      Response,
+      Params,
+      TransformedResponse,
+      boolean | undefined
+    >,
+    'onChange'
+  > {
+  onChange: (_: SyntheticEvent, value: TransformedResponse | null) => void
+}
+
+const DividedDropdownAutocomplete = <
+  Response,
+  TransformedResponse,
+  Params = undefined
+>({
+  axiosProps,
+  groupBy,
+  ...props
+}: DividedDropdownAutocompleteProps<Response, TransformedResponse, Params>) => {
+  const { t } = useTranslation()
+
+  const optionsList = (params: AutocompleteRenderGroupParams) => {
+    const { group, key, children, ...props } = params
+
+    return (
+      <Box key={key} {...props}>
+        {Number(key) > 0 && <Divider sx={styles.autocompleteDropdownDivider} />}
```

Comment:
Maybe just use `&&` operator. in our case `> 0` it's also truthy

Reply:
I guess it should be > 0, if I will use only && it will also render divider when key is equal to 0